pkg/infra/usagestats/statscollector/prometheus_flavor.go

```diff
@@ -0,0 +1,129 @@
+package statscollector
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+const promFlavorCacheLifetime = time.Hour
+
+type memoPrometheusFlavor struct {
+	variants map[string]int64
+
+	memoized time.Time
+}
+
+func (s *Service) detectPrometheusVariants(ctx context.Context) (map[string]int64, error) {
+	if s.promFlavorCache.memoized.Add(promFlavorCacheLifetime).After(time.Now()) &&
+		s.promFlavorCache.variants != nil {
+		return s.promFlavorCache.variants, nil
+	}
+
+	dsProm := &models.GetDataSourcesByTypeQuery{Type: "prometheus"}
+	err := s.datasources.GetDataSourcesByType(ctx, dsProm)
+	if err != nil {
+		s.log.Error("Failed to read all Prometheus data sources", "error", err)
+		return nil, err
+	}
+	group, ctx := errgroup.WithContext(ctx)
+
+	variantChan := make(chan string)
+	for _, ds := range dsProm.Result {
+		ds := ds
+		group.Go(func() error {
```

Comment:
I'd avoid go routines to reduce complexity a little bit. This runs ever 24h. It does not matter if it takes 5min more to query all datasources :)

Reply:
:+1: I'll revert back to a naive implementation.

It's not _quite_ that straightforward, since this is used by licensing as well, but still, it shouldn't be a problem.